DuckDuckGo/MainViewController.swift

```diff
@@ -392,8 +399,8 @@ class MainViewController: UIViewController {
 
     fileprivate func displayAutocompleteSuggestions(forQuery query: String) {
         if autocompleteController == nil && appSettings.autocomplete {
-            allowContentUnderflow = false
             let controller = AutocompleteViewController.loadFromStoryboard()
+            controller.shouldOffsetY = allowContentUnderflow
```

Comment:
Originally I had it set `allowContentUnderflow = false` when the autocomplete was shown, but there was an easy to replicate bug:

1. With centered search on the home page, enter a search term that shows autocomplete results
1. Tap the screen to exit autocomplete / close the keyboard
1. The fake search field would then be visible

I had a hack which set `allowContentUnderflow = true` but the keyboard animation made it look really obvious and I couldn't work out how to hide it without it being see by the user.

This approach tells the autocomplete view controller to use a vertical offset or not depending on the current state.

Happy to hear other suggestions for this!

Reply:
No reasonable alternative has came to my mind so far. :)